src/Tests/Microsoft.Diagnostics.Monitoring.TestCommon/TaskExtensions.cs

```diff
@@ -40,5 +41,28 @@ public static async Task<T> SafeAwait<T>(this Task<T> task, ITestOutputHelper ou
             }
             return fallbackValue;
         }
+
+        public static async Task WithCancellation(this Task task, CancellationToken token)
+        {
+            CancellationTokenSource localCancellationTokenSource = new();
+            CancellationTokenSource combinedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(token, localCancellationTokenSource.Token);
+
+            try
+            {
+                // Creating a Task.Delay with Infinite timeout is not intended to work with a cancellation token that never fires
+                // So we need to make a local "combined" token that we can cancel if the provided CancellationToken token is never canceled
+                // This allows the framework to properly cleanup the Task and it's associated token registrations
+                Task waitOnCancellation = Task.Delay(Timeout.Infinite, combinedTokenSource.Token);
+                await Task.WhenAny(task, waitOnCancellation).ConfigureAwait(false);
```

Comment:
This will cause a TaskCancelledException to be thrown when either task is cancelled.
```suggestion
                await Task.WhenAny(task, waitOnCancellation).Unwrap().ConfigureAwait(false);
```